src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -274,9 +275,24 @@ static int createSoftKernelFile(uint64_t paddr, size_t size, uint32_t cuidx)
   return 0;
 }
 
+static void stacktrace_logger(const int sig)
+{
+  syslog(LOG_ERR, "%s - got %d\n", __func__, sig);
+  if (sig == SIGCHLD)
+    return;
+  void * array[25];
+  int nSize = backtrace(array, 25);
+  char ** symbols = backtrace_symbols(array, nSize);
+  for (int i = 0; i < nSize; i++)
```

Comment:
You may want to check if symbols is valid in case of backtrace_symbols is running out of time to avoid crash in crash handler. :)

Reply:
Reminds of XSIM simulator back trace printer I had implemented many years ago.
Please add additional checks on backtrace() and backtrace_symbols() as suggested.